fastai/learner.py

```diff
@@ -272,17 +276,17 @@ def validate(self, ds_idx=1, dl=None, cbs=None):
 
     @delegates(GatherPredsCallback.__init__)
     def get_preds(self,
-                  ds_idx: int =1, # This takes the dataset index of DataLoader with default value as 1 for valid and 0 can be used for train
-                  dl: TfmdDL | None=None, # `DataLoaders` containing data for each dataset needed for `model`
-                  with_input: bool=False, # Whether to return inputs
-                  with_decoded: bool=False, # Whether to decode based on loss function passed
-                  with_loss : bool=False, # Whether to return losses
-                  act: Any=None, # Option to pass Activation function to predict function
-                  inner : bool=False, # Tells that it's used internally used anywhere like in another training loop
-                  reorder : bool=True, # To order the tensors appropriately
-                  cbs: list | None =None, # Option to pass `Callbacks` to predict function
-                  **kwargs
-                 )-> tuple:
+        ds_idx:int=1, # `DataLoader` to use for predictions if `dl` is None. 0: train. 1: valid
+        dl:DataLoader|None=None, # `DataLoader` to use for predictions, defaults to `ds_idx=1` if None
```

Comment:
I suspect other iterables might work - e.g. a progress bar wrapped DL?

Reply:
Not sure what this should to be changed to. `DataLoader` seems to be useful information to have in the type hint.

Reply:
The types aren't really just hints - they have to handle at minimum all the types that can be accepted, otherwise people that use them will get errors. In this case, I believe the only type that would be accurate is `iterable` -- or just have no type, in which case it defaults to `Any`.